docs/Proxies.md

```diff
@@ -0,0 +1,146 @@
+
+
+### Implementation contract
+
+The implementation contract, also known as the logic contract, receives the redirected function calls from the proxy contract. The implementation contract should follow the [Extensibility pattern](../docs/Extensibility.md#the-pattern) and import directly from the [Proxy library](../openzeppelin/upgrades/library.cairo).
+ 
+
+The implementation contract should:
+- import `Proxy_initializer` and `Proxy_set_implementation`
+- initialize the proxy immediately after contract deployment.
+
+If the implementation is upgradeable, it should:
+- include a method to upgrade the implementation (i.e. `upgrade`)
+- use access control to protect the contract's upgradeability.
+
+The implementation contract should NOT:
+- deploy with a traditional constructor—use an initializer method instead that invokes `Proxy_initializer` and the implementation contract's initializer
+
+> Note that the imported `Proxy_initializer` includes a check the ensures the initializer can only be called once; however, `Proxy_set_implementation` does not include this check. It's up to the developers to protect their implementation contract's upgradeability with access controls such as `Proxy_only_admin`. 
+
+For a full implementation contract example, please see:
+- [Proxiable implementation](../tests/mocks/proxiable_implementation.cairo)
+
+## Using proxies 
+
+### Contract upgrades
+
+To upgrade a contract, the implementation contract should include an `upgrade` method that, when called, changes the reference to a new deployed contract like this:
+
+
+```python
+    # deploy first implementation
+    IMPLEMENTATION = await starknet.deploy(
+        "path/to/implementation.cairo",
+        constructor_calldata=[]
+    )
+
+    # deploy proxy
+    PROXY = await starknet.deploy(
+        "path/to/proxy.cairo",
+        constructor_calldata=[
+            IMPLEMENTATION.contract_address,  # set implementation address
+        ]
+    )
+
+    # deploy implementation v2
+    IMPLEMENTATION_V2 = await starknet.deploy(
+        "path/to/implementation_v2.cairo",
+        constructor_calldata=[]
+    )
+
+    # call upgrade with the new implementation contract address
+    await signer.send_transaction(
+        account, PROXY.contract_address, 'upgrade', [
+            IMPLEMENTATION_V2.contract_address
+        ]
+    )
+```
+
+For a full deployment and upgrade implementation, please see:
+- [Upgrades V1](../tests/mocks/upgrades_v1_mock.cairo)
+- [Upgrades V2](../tests/mocks/upgrades_v2_mock.cairo)
+
+### Handling method calls
+
+As with most StarkNet contracts, interacting with a proxy contract requires an [account abstraction](../docs/Account.md#quickstart). One notable difference with proxy contracts versus other contract implementations is that calling `@view` methods also requires an account abstraction. The differences written in Python, for example, are as follows:
```

Comment:
Why is it mandatory? What happens if I try to `starknet invoke` directly to a proxy?

Reply:
Without an account contract, method calls/invocations return an `AttributeError`. I'll add that to the docs